### DISCUSSION TITLE
Create LIST column from MAP column with value as occurrences

### DISCUSSION THREAD

**Discussion by muralisoundararajan** [duckdb/duckdb#13919] [Q&A] [answered]
Hi All,

I have a large table which is similar to the below table where there is a column `Response Times` which is a map of the `Response Time` and `Occurrences`

| Customer    | Response Times    |
| ------------- | ------------- |
| AB | {0.001: 1, 0.020: 2} |
| CD | {0.003: 2, 0.010: 3} |
| EF | {0.021: 2, 0.040: 2} |

Are there any other methods available to create the following table without creating a separate row for each occurrence (`CROSS JOIN` on `RANGE` of occurrences)?

| Customer    | Response Time    |
| ------------- | ------------- |
| AB | [0.001, 0.020, 0.020] |
| CD | [0.003, 0.003, 0.010, 0.010, 0.010] |
| EF | [0.021,0.021, 0.040, 0.040] |

Regards,
Murali Soundararajan

**Comment by Tishj** [comment, marked as answer]
This should work:
```sql
select 
	Customer,
	flatten([
		([x.key for y in range(x.value)])
		for
			x
		in map_entries("Response Times")
	]) as "Response Times"
from tbl;
```

**Comment by muralisoundararajan** [comment]
Thanks for the quick solution @Tishj
